aip/auth/4110.md

```diff
@@ -136,25 +136,29 @@ digraph d_front_back {
     1. Otherwise, get an access token _[END]_
   1. If false, go to step (2.3)
 1. **Determine auth flows**
-  1. If the credential is gcloud credential go to step (5.3)
-  1. If target audience or scope is provided by the developer go to step (5.1)
+  1. If the credential is a Gcloud credential JSON go to step (5.3)
+  1. If the credential is a service account key JSON:
+     1. If neither `scope` or `audience` is available, return an error _[END]_
+     1. If [scope is supported for self-signed JWT](8), go to step (5.1)
+     1. If `audiences` are **explicitly** provided by developer, go to step (5.1)
+     1. If `scope` is not available, go to step (5.1)
+     1. Otherwise, go to step (5.2)
```

Comment:
I'm not sure I understand when this otherwise would be triggered, it might be more clear if you replace "If [scope is supported for self-signed JWT](8), go to step (5.1)" with "If scope is available and [scope is not supported for self-signed JWT](8), go to step (5.2), then i think you can collapse the rest into "Otherwise, go to step (5.2)" and have the branches for the different cases without somewhat repeting them.

Reply:
I also feel like "If scope is supported for self-signed JWT" is an ambiguous condition to put upfront - this is a moot point if scope is not available. My suggestion is the follow arrangement:
1. If neither `scope` or `audience` is available, return an error _[END]_
2. If scope is not available, go to step (5.1)
3. If scope is available and EnableJWTAccessWithScope is true, go to step (5.1)
4. If audiences are explicitly provided, go to step (5.1)
5. Otherwise, go to step (5.2)
Silvano's suggestion could work too though the redirection would be out of order (5.2,5.1,5.1,5.2)

Reply:
I have re-ordered this a bit hopefully its easier to follow this time. Thanks.